backend/spec/controllers/spree/admin/users_controller_spec.rb

```diff
@@ -59,6 +59,33 @@
         expect(assigns(:collection)).to eq [user]
       end
     end
+
+    context "when Spree.user_class have a different namespace than Spree" do
+      class UserModel < ApplicationRecord
+        self.table_name = 'spree_users'
+        include Spree::UserMethods
+      end
+
+      around do |example|
+        Spree.user_class = 'UserModel'
+        UserModel.create(email: "a@solidus.io")
+        example.run
+        Spree.user_class = 'Spree::LegacyUser'
+      end
```

Comment:
Can we take the old value of `user_class` and reset it later instead of hardcoding the value that we are using now? I.e.:


```suggestion
        actual_user_class = Spree.user_class
        Spree.user_class = 'UserModel'
        UserModel.create(email: "a@solidus.io")
        example.run
        Spree.user_class = actual_user_class
      end
```